### PR TITLE
UWC : Rebased UWC to EII pv

### PR DESCRIPTION
**Pull request by aditisa1**
1) Updated Dockerfile to include snyk fix for apt-get install
2) updated 01 script
3) added rm cmd in 02 script

Signed-off-by: aditisa1 <aditi.saxena@intel.com>

### CODE REVIEW

File: kpi-tactic/Dockerfile [hunk 1]
```diff
@@ -21,13 +21,35 @@
...
+    mkdir -p deps && \
+    cd deps && \
+    wget -q --show-progress https://github.com/DaveGamble/cJSON/archive/v1.7.12.tar.gz -O cjson.tar.gz && \
+    tar xf cjson.tar.gz && \
+    cd cJSON-1.7.12 && \
+    mkdir build && cd build && \
+    cmake -DCMAKE_INSTALL_INCLUDEDIR=${CMAKE_INSTALL_PREFIX}/include -DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX} .. && \
+    make install
 
+RUN rm -rf /var/lib/apt/lists/*
```
Comment (nagdeepgk) [inline kpi-tactic/Dockerfile:52]: Why was this add now ? Something changed in EII?

File: kpi-tactic/Dockerfile [hunk 2]
```diff
@@ -44,7 +66,8 @@ COPY --from=uwc_common /uwc_util/Release/libuwc-common.so /KPIApp/lib/
 COPY --from=uwc_common /uwc_util/include/* /KPIApp/include/
 COPY --from=uwc_common $CMAKE_INSTALL_PREFIX/include /usr/local/include
 COPY --from=uwc_common $CMAKE_INSTALL_PREFIX/lib /usr/local/lib
-
+COPY --from=uwc_common ${CMAKE_INSTALL_PREFIX}/include ${CMAKE_INSTALL_PREFIX}/include
```
Comment (nagdeepgk) [inline kpi-tactic/Dockerfile:69]: Why was this add now ? Something changed in EII?

File: kpi-tactic/Dockerfile [hunk 3]
```diff
@@ -53,17 +76,18 @@ RUN cd /KPIApp/Release \
...
 	&& mkdir -p /opt/intel/app/logs \
 	&& mkdir -p /opt/intel/config \
 	&& mkdir -p /opt/intel/eii/uwc_data
 WORKDIR ${APP_WORK_DIR}
-
-RUN apt update && apt install --no-install-recommends -y libcjson1 libzmq5 zlib1g && \
-    rm -rf /var/lib/apt/lists/*
+ARG CMAKE_INSTALL_PREFIX
 # copy required dependencies
+COPY --from=kpitactic ${CMAKE_INSTALL_PREFIX}/lib  ${APP_WORK_DIR}
```
Comment (nagdeepgk) [inline kpi-tactic/Dockerfile:89]: Why was this add now ? Something changed in EII?

File: modbus-master/Dockerfile [hunk 4]
```diff
@@ -72,13 +96,12 @@ RUN mkdir -p /opt/intel/app \
...
 	&& mkdir -p /opt/intel/eii/uwc_data
 
-RUN apt update && apt install --no-install-recommends -y libcjson1 libzmq5 zlib1g && \
-    rm -rf /var/lib/apt/lists/*
-
 WORKDIR ${APP_WORK_DIR}
 
 # copy required libs and includes
 ARG CMAKE_INSTALL_PREFIX
+COPY --from=modbus-app ${CMAKE_INSTALL_PREFIX}/lib  ${APP_WORK_DIR}
```
Comment (nagdeepgk) [inline modbus-master/Dockerfile:103]: Why was this add now ? Something changed in EII?

File: modbus-master/Dockerfile_RTU [hunk 5]
```diff
@@ -21,6 +21,7 @@
 ARG EII_VERSION
 ARG UBUNTU_IMAGE_VERSION
 ARG UWC_VERSION
+ARG CMAKE_INSTALL_PREFIX
```
Comment (nagdeepgk) [inline modbus-master/Dockerfile_RTU:24]: Why was this add now ? Something changed in EII?

File: mqtt-bridge/Dockerfile [hunk 6]
```diff
@@ -42,15 +64,15 @@ COPY --from=uwc_common /EII/log4cpp/build/include/log4cpp /MQTT-Bridge/include/l
 COPY --from=uwc_common /EII/log4cpp/build/lib/* /MQTT-Bridge/lib/
 COPY --from=uwc_common /uwc_util/Release/libuwc-common.so /MQTT-Bridge/lib/
 COPY --from=uwc_common /uwc_util/include/* /MQTT-Bridge/include/
-ARG CMAKE_INSTALL_PREFIX
-COPY --from=uwc_common $CMAKE_INSTALL_PREFIX/include /usr/local/include
-COPY --from=uwc_common $CMAKE_INSTALL_PREFIX/lib /usr/local/lib
+COPY --from=uwc_common ${CMAKE_INSTALL_PREFIX}/include /usr/local/include
+COPY --from=uwc_common ${CMAKE_INSTALL_PREFIX}/lib /usr/local/lib
```
Comment (nagdeepgk) [inline mqtt-bridge/Dockerfile:68]: Why was this add now ? Something changed in EII?

File: mqtt-bridge/Dockerfile [hunk 7]
```diff
@@ -61,22 +83,20 @@ RUN mkdir -p /opt/intel/app \
...
 	&& mkdir -p /opt/intel/eii/uwc_data
 
-RUN apt update && apt install --no-install-recommends -y libcjson1 libzmq5 zlib1g && \
-    rm -rf /var/lib/apt/lists/*
-
 WORKDIR ${APP_WORK_DIR}
 # copy required dependencies
-COPY --from=mqtt-bridge /MQTT-Bridge/Release/MQTT_Bridge ${APP_WORK_DIR}
+ARG CMAKE_INSTALL_PREFIX
+COPY --from=builder ${CMAKE_INSTALL_PREFIX}/lib  ${APP_WORK_DIR}
```
Comment (nagdeepgk) [inline mqtt-bridge/Dockerfile:89]: Why was this add now ? Something changed in EII?

File: sparkplug-bridge/Dockerfile [hunk 8]
```diff
@@ -75,12 +97,11 @@ RUN mkdir -p /opt/intel/app \
...
 	&& mkdir -p /opt/intel/eii/uwc_data
 
-RUN apt update && apt install --no-install-recommends -y libcjson1 libzmq5 zlib1g && \
-    rm -rf /var/lib/apt/lists/*
-
 WORKDIR ${APP_WORK_DIR}
-
+ARG CMAKE_INSTALL_PREFIX
 # copy required dependencies
+COPY --from=sparkplug-bridge ${CMAKE_INSTALL_PREFIX}/lib  ${APP_WORK_DIR}
```
Comment (nagdeepgk) [inline sparkplug-bridge/Dockerfile:103]: Why was this add now ? Something changed in EII?

File: uwc_common/Dockerfile.common [hunk 9]
```diff
@@ -29,7 +29,8 @@ FROM ia_common:$EII_VERSION as common
 # build openssl library
 FROM ia_common:$EII_VERSION as openssl-build
 ARG OPENSSL_DIR=openssl
-
+RUN apt-get update && \
```
Comment (nagdeepgk) [inline uwc_common/Dockerfile.common:32]: Why was this add now ? Something changed in EII?

File: Vendor_Apps/EmbPublisher/Dockerfile [hunk 10]
```diff
@@ -46,7 +69,7 @@ ENV PATH="$PATH:/usr/local/go/bin" \
     LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:${CMAKE_INSTALL_PREFIX}/lib"
 
 # These flags are needed for enabling security while compiling and linking with cpuidcheck in golang
-ENV CGO_CFLAGS="$CGO_FLAGS -I ${CMAKE_INSTALL_PREFIX}/include -O2 -D_FORTIFY_SOURCE=2 -Werror=format-security -fstack-protector-strong -fPIC" \
+ENV CGO_CFLAGS="$CGO_FLAGS -I ${CMAKE_INSTALL_PREFIX}/include -O2 -D_FORTIFY_SOURCE=2 -Werror=format-security -fstack-protector-strong -fno-strict-overflow -fno-delete-null-pointer-checks -fwrapv -fPIC" \
```
[the remark this answers is not in the document]
Comment (aditisa1) [reply Vendor_Apps/EmbPublisher/Dockerfile:72]: yes nagdeep 
taken from EII only

File: Vendor_Apps/EmbPublisher/Dockerfile [hunk 11]
```diff
@@ -61,8 +84,7 @@ ARG EII_UID
 ARG EII_USER_NAME
 RUN groupadd $EII_USER_NAME -g $EII_UID && \
     useradd -r -u $EII_UID -g $EII_USER_NAME $EII_USER_NAME
-RUN apt update && apt install --no-install-recommends -y libcjson1 libzmq5 zlib1g && \
```
[the remark this answers is not in the document]
Comment (aditisa1) [reply Vendor_Apps/EmbPublisher/Dockerfile:64]: as the above changes are made in EII too as part of snyk changes

File: Vendor_Apps/EmbPublisher/pub_tool.go [hunk 12]
```diff
@@ -63,7 +62,7 @@ func main() {
 		return
 	}
 	msg_file := "/datafiles/" + appConfig["msg_file"].(string)
-	msg, err := eiimsgbus.ReadJsonConfig(msg_file)
+	msg, err := eiimsgbus.ReadJSONConfig(msg_file)
```
[the remark this answers is not in the document]
Comment (aditisa1) [reply Vendor_Apps/EmbPublisher/pub_tool.go:65]: yess

File: Vendor_Apps/EmbSubscriber/Dockerfile [hunk 13]
```diff
@@ -31,22 +31,43 @@ LABEL description="Emb Subscriber image"
...
-
 COPY ./*.go ./
 COPY go.mod ./
 ENV PATH="$PATH:/usr/local/go/bin" \
     PKG_CONFIG_PATH="$PKG_CONFIG_PATH:${CMAKE_INSTALL_PREFIX}/lib/pkgconfig" \
     LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:${CMAKE_INSTALL_PREFIX}/lib"
 
 # These flags are needed for enabling security while compiling and linking with cpuidcheck in golang
-ENV CGO_CFLAGS="$CGO_FLAGS -I ${CMAKE_INSTALL_PREFIX}/include -O2 -D_FORTIFY_SOURCE=2 -Werror=format-security -fstack-protector-strong -fPIC" \
+ENV CGO_CFLAGS="$CGO_FLAGS -I ${CMAKE_INSTALL_PREFIX}/include -O2 -D_FORTIFY_SOURCE=2 -Werror=format-security -fstack-protector-strong -fno-strict-overflow -fno-delete-null-pointer-checks -fwrapv -fPIC" \
```
[the remark this answers is not in the document]
Comment (aditisa1) [reply Vendor_Apps/EmbSubscriber/Dockerfile:70]: taken from EII only

File: build_scripts/01_uwc_pre_requisites.sh [hunk 14]
```diff
@@ -187,12 +187,17 @@ network_setup()
...
 #------------------------------------------------------------------
 modify_env(){
+	cd ${working_dir}
 	file="${working_dir}/../eii_configs/envs.csv"
 	eCollection=( $(cut -d ',' -f2 $file ) )
-	sed -i 's/'ETCDROOT_PASSWORD=\'\''/'${eCollection[0]}'/g' ${working_dir}/../../build/.env
-	sed -i 's/'INFLUXDB_USERNAME=\'\''/'${eCollection[1]}'/g' ${working_dir}/../../build/.env
-	sed -i 's/'INFLUXDB_PASSWORD=\'\''/'${eCollection[2]}'/g' ${working_dir}/../../build/.env
-
+	result=`grep -Fi ${eCollection[0]} ../../build/.env`
```
[the remark this answers is not in the document]
Comment (aditisa1) [reply build_scripts/01_uwc_pre_requisites.sh:193]: To check if the env are already updated 
if it is updated then not to make change

File: build_scripts/02_provision_build_UWC.sh [hunk 15]
```diff
@@ -40,6 +40,7 @@ brokerAddr=""
 brokerPort=""
 qos=""
 ret=""
+emb_sample_modules=0
```
[the remark this answers is not in the document]
Comment (aditisa1) [reply build_scripts/02_provision_build_UWC.sh:43]: To distinguish between tools/embpublihser and one Vendorapp one

File: kpi-tactic/Dockerfile [hunk 16]
```diff
@@ -21,13 +21,35 @@
 ARG EII_VERSION
 ARG UBUNTU_IMAGE_VERSION
 ARG UWC_VERSION
+ARG CMAKE_INSTALL_PREFIX
```
[the remark this answers is not in the document]
Comment (aditisa1) [reply kpi-tactic/Dockerfile:24]: yess
for synk changes made by EII
changed it's position

File: kpi-tactic/Dockerfile (hunk 1 above)
Comment (aditisa1) [reply kpi-tactic/Dockerfile:52]: yess
part of same change

File: kpi-tactic/Dockerfile (hunk 3 above)
Comment (aditisa1) [reply kpi-tactic/Dockerfile:89]: To copy the libs added in build stage(kpitactic) to run stage

File: modbus-master/Dockerfile_RTU (hunk 5 above)
Comment (aditisa1) [reply modbus-master/Dockerfile_RTU:24]: yess
part of same change

File: mqtt-bridge/Dockerfile (hunk 6 above)
Comment (aditisa1) [reply mqtt-bridge/Dockerfile:68]: Nothing changed here 
same lines only

File: sparkplug-bridge/Dockerfile (hunk 8 above)
Comment (aditisa1) [reply sparkplug-bridge/Dockerfile:103]: To copy the libs added in build stage(sparkplug-bridge) to run stage

File: uwc_common/Dockerfile.common (hunk 9 above)
Comment (aditisa1) [reply uwc_common/Dockerfile.common:32]: yess
as we need wget in build stage for installation of libs

**Pull request closed without merging**